packages/python/plotly/plotly/tests/test_core/test_px/test_px_input.py

```diff
@@ -0,0 +1,144 @@
+import plotly.express as px
+import numpy as np
+import pandas as pd
+import pytest
+import plotly.graph_objects as go
+import plotly
+from plotly.express._core import build_or_augment_dataframe
+from pandas.util.testing import assert_frame_equal
+
+attrables = (
+    ["x", "y", "z", "a", "b", "c", "r", "theta", "size", "dimensions"]
+    + ["custom_data", "hover_name", "hover_data", "text"]
+    + ["error_x", "error_x_minus"]
+    + ["error_y", "error_y_minus", "error_z", "error_z_minus"]
+    + ["lat", "lon", "locations", "animation_group"]
+)
+array_attrables = ["dimensions", "custom_data", "hover_data"]
+group_attrables = ["animation_frame", "facet_row", "facet_col", "line_group"]
+
+all_attrables = attrables + group_attrables + ["color"]
+
+
+def test_numpy():
+    fig = px.scatter(x=[1, 2, 3], y=[2, 3, 4], color=[1, 3, 9])
+
+
+def test_numpy_labels():
+    fig = px.scatter(
+        x=[1, 2, 3], y=[2, 3, 4], labels={"x": "time"}
+    )  # other labels will be kw arguments
+    assert fig.data[0]["hovertemplate"] == "time=%{x}<br>y=%{y}"
+
+
+def test_with_index():
+    tips = px.data.tips()
+    fig = px.scatter(tips, x=tips.index, y="total_bill")
+    fig = px.scatter(tips, x=tips.index, y=tips.total_bill)
+    fig = px.scatter(tips, x=tips.index, y=tips.total_bill)
+    assert fig.data[0]["hovertemplate"] == "index=%{x}<br>total_bill=%{y}"
+    # I was not expecting this to work but it does...
+    fig = px.scatter(tips, x=tips.index, y=10 * tips.total_bill)
+    assert fig.data[0]["hovertemplate"] == "index=%{x}<br>total_bill=%{y}"
```

Comment:
wow, I'm surprised that this works ... how does it work with the `total_bill` name??

Reply:
Oh I see. scary :)

Reply:
For the docstrings of `px` functions I chose to use the name `array_like` which is commonly used in numpy to refer to both arrays and lists, and in the broad sense dataframes as well. This is hopefully ready for another round of review.

Reply:
once black is happy of course...